extension/src/background/helpers/dataStorage.ts

```diff
@@ -13,37 +13,60 @@ interface SetItemParams {
   [key: string]: any;
 }
 
-export const browserStorage = browser.storage?.local;
+// https://github.com/mozilla/webextension-polyfill/issues/424
+interface BrowserStorage extends browser.Storage.Static {
+  session: browser.Storage.LocalStorageArea;
+}
+
+const storage = browser.storage as BrowserStorage;
+
+// browser storage uses local storage which stores values on disk and persists data across sessions
+// session storage uses session storage which stores data in memory and clears data after every "session"
+// only use session storage for secrets or sensitive values
+export const browserStorage = storage?.local;
+export const sessionStorage = storage?.session;
+
+// Session Storage Feature Flag - turn on when storage.session is supported
+export const SESSION_STORAGE_ENABLED = false;
 
-export const dataStorage = {
+export type StorageOption = typeof browserStorage | typeof sessionStorage;
+
+export const dataStorage = (storageApi: StorageOption = browserStorage) => ({
   getItem: async (key: string) => {
     // TODO: re-enable defaults by passing an object. The value of the key-value pair will be the default
 
-    const storageResult = await browser.storage.local.get(key);
+    const storageResult = await storageApi.get(key);
 
     return storageResult[key];
   },
   setItem: async (setItemParams: SetItemParams) => {
-    await browser.storage.local.set(setItemParams);
+    await storageApi.set(setItemParams);
   },
 
   clear: async () => {
-    await browser.storage.local.clear();
+    await storageApi.clear();
   },
-};
+});
 
-export const dataStorageAccess = {
-  getItem: (keyId: string) => dataStorage.getItem(keyId),
-  setItem: async (keyId: string, value: any) => {
-    await dataStorage.setItem({ [keyId]: value });
-  },
-  clear: () => dataStorage.clear(),
+export const dataStorageAccess = (
+  storageApi: StorageOption = browserStorage,
+) => {
+  const store = dataStorage(storageApi);
+  return {
+    getItem: store.getItem,
+    setItem: async (keyId: string, value: any) => {
+      await store.setItem({ [keyId]: value });
+    },
+    clear: () => store.clear(),
+  };
 };
 
 // This migration adds a friendbotUrl to testnet and futurenet network details
 export const migrateFriendBotUrlNetworkDetails = async () => {
+  const store = dataStorageAccess(browserStorage);
```

Comment:
for ex: we can call this `const localStore = dataStorage(localStorage)` to make it clear that these methods are saving to the long lived store